pkg/scheduler/store.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"sync"
+)
+
+/*
+The relationship between relevant API objects (arrows indicate selections/references):
+ep -> loc -> st
+ep: edgeplacement; loc: location; st: synctarget
+
+To facilitate incremental updates, we need to maintain internal data structures:
+
+	ep <- loc <- st
+
+or in other words,
+
+	ep <- loc
+	loc <- st
+	ep <- st
+
+*/
+
+type internalData struct {
+	l sync.Mutex
+
+	// TODO(waltforme): should these map values be slices or maps?
+	epsBySelectedLoc map[string][]string // ep <- loc
```

Comment:
I think you could make this field name shorter, because there are no other Locations relevant to an EdgePlacement besides the ones that it selects.  You could omit "Selected" from this field name.

An analogous comment applies to the other fields here too.

Reply:
Technically it is correct that the only relationship is "selection". But I'd like to keep the names here, because they reads more natural and carries more essential info, thus make the thinking easier when I use the names.